// azdata-test/out/stubs/modelView/stubComponent.js
"use strict";
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the Source EULA. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.StubComponent = void 0;
var StubComponent = /** @class */ (function () {
    function StubComponent() {
        this.height = 0;
        this.width = 0;
        this.position = 'initial';
        this.enabled = false;
        this.display = 'initial';
        this.ariaLabel = '';
        this.ariaRole = '';
        this.ariaSelected = false;
        this.CSSStyles = {};
    }
    StubComponent.prototype.updateProperties = function (properties) {
        throw new Error('Method not implemented.');
    };
    StubComponent.prototype.updateProperty = function (key, value) {
        throw new Error('Method not implemented.');
    };
    StubComponent.prototype.updateCssStyles = function (cssStyles) {
        throw new Error('Method not implemented.');
    };
    StubComponent.prototype.validate = function () {
        throw new Error('Method not implemented.');
    };
    StubComponent.prototype.focus = function () {
        throw new Error('Method not implemented.');
    };
    return StubComponent;
}());
exports.StubComponent = StubComponent;
